apiserver/facades/client/bundle/mock_test.go

```diff
@@ -5,10 +5,48 @@ package bundle_test
 
 import (
 	"github.com/juju/testing"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/description"
+	"github.com/juju/juju/apiserver/facades/client/bundle"
+	"github.com/juju/juju/state"
 )
 
 type mockState struct {
 	testing.Stub
+	bundle.Backend
+}
+
+func (m *mockState) SetExportconfig(cfg state.ExportConfig) {
+	cfg.SkipActions = true
+	cfg.SkipCloudImageMetadata = true
+	cfg.SkipCredentials = true
+	cfg.SkipIPAddresses = true
+	cfg.SkipSSHHostKeys = true
+	cfg.SkipStatusHistory = true
+	cfg.SkipLinkLayerDevices = true
+}
+
+func (m *mockState) ExportpartialNoApplication(config state.ExportConfig) (description.Model, error) {
```

Comment:
This does not look right.

For one mock state can only implement the methods from an actual state.
There are couple of other things here. Let's talk through them f-2-f.